Fix 16993 Chromevox is not reading learning progress instead it is reading numbers

Uglycrown opened the pull request:
## Overview

1. This PR fixes #16993  ..
2. This PR does the following: Now Chromevox is reading learning progress percentage (%) instead reading numbers
## Essential Checklist.

- [x]  The PR title starts with "Fix #16993 .: ", followed by a short, clear summary of the changes. [lesson-information-card-modal.component.html](https://github.com/oppia/oppia/blob/develop/core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.html)
 [lesson-information-card-modal.component.ts](https://github.com/oppia/oppia/blob/develop/core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.ts)
[lesson-information-card-modal.component.spec.ts](https://github.com/oppia/oppia/blob/develop/core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.spec.ts)
- [x]  The linter/Karma presubmit checks have passed locally on your machine.
- [x]  "Allow edits from maintainers" is checked.
- [x]  The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct

**After Changes**

https://user-images.githubusercontent.com/87015615/232764237-2ea676bf-255d-459b-8419-feccee34e25c.mp4

Uglycrown committed:
Now Chromevox can read close button for x icon after uploading a picture

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
label-fix-chromvox

Uglycrown committed:
fix-ffa

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
Merge remote-tracking branch 'upstream/develop' into develop

Uglycrown committed:
fix-read-percentage-lesson-progress

Uglycrown committed:
fix-read-percentage-lesson-progress

Uglycrown committed:
progress-bar-fix

Uglycrown committed:
fix-progress

Uglycrown committed:
fix-progress

Uglycrown committed:
Progress-percentage-fix

Uglycrown committed:
fix-progressbar-read

Uglycrown committed:
fix-progress-bar-reading-issue

Uglycrown committed:
fix-progress-bar-reading-issue

Uglycrown committed:
fix-progress-bar-reading-issue

Uglycrown committed:
fix-progress-bar-reading-issue

Uglycrown committed:
proessbar-read-chromevox-fix

Uglycrown committed:
proessbar-read-chromevox-fix

Uglycrown committed:
progressbar-read-fix

Uglycrown committed:
progressbar-read-fix

Uglycrown committed:
progressbar-read-fix

Uglycrown committed:
progressbar-read-fix

Uglycrown committed:
progressbar-read-fix

Uglycrown committed:
progressbar-read-fix

Uglycrown committed:
progressbar-fix

Uglycrown committed:
progressbar-fix

Uglycrown committed:
progressbar-read-fix

Uglycrown committed:
progressbar-read-fix

Uglycrown committed:
progressbar-read-fix

Uglycrown committed:
progressbar-read-fix

Uglycrown committed:
progressbar-read-fix

Uglycrown committed:
progressbar-read-fix

Uglycrown committed:
progressbar-read-fix

Uglycrown committed:
progressbar-read-fix

Uglycrown committed:
Merge branch 'develop' into progressbar-read-fix

Uglycrown commented:
Hi, @seanlip just opened PR please let me know if any changes are required into this.

seanlip commented:
Thanks --- no concerns with the UI. Please feel free to assign reviewers!

seanlip commented:
@Uglycrown Also, next time you have a question for me, please make sure I'm assigned to the PR. Otherwise I might not see it.

In general, you'll want to ensure that the Assignees list is correct if you want folks to take a look at things.

Uglycrown committed:
Merge branch 'develop' into progressbar-read-fix

gp201 commented on core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.spec.ts:
```diff
@@ -436,4 +436,39 @@ describe('Lesson Information card modal component', () => {
+  it('should return 0% when no checkpoints are completed', () => {
+    componentInstance.completedCheckpointsCount = 0;
+    componentInstance.checkpointCount = 5;
+    const progressPercentage = componentInstance.getProgressPercentage();
+    expect(progressPercentage).toEqual('0');
```
Can we combine this into a single line?

Please do the same for the remaining tests

Uglycrown replied:
done.

gp201 commented on core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.spec.ts:
```diff
@@ -436,4 +436,39 @@ describe('Lesson Information card modal component', () => {
+
+  it('should return 0% when no checkpoints are completed', () => {
+    componentInstance.completedCheckpointsCount = 0;
+    componentInstance.checkpointCount = 5;
```
Add a new line after this line. Please do the same for the remaining tests

Uglycrown replied:
done.

gp201 commented on core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.spec.ts:
```diff
@@ -436,4 +436,39 @@ describe('Lesson Information card modal component', () => {
+    expect(progressPercentage).toEqual('25');
+  });
+
+  it('should return the correct percentage for 75% progress', () => {
```
Do we need to test 25% and 75%?

Uglycrown replied:
@gp201, It is not necessary but it is good if at least one test is present in them. I can remove both tests if you want. let me know what should I do.

gp201 replied:
Let's have the following tests:

- 25%
- rounds to the near number (67?)
- 100%
- 0%

gp201 reviewed (changes requested):
@Uglycrown Nice work! Can you change the PR title to briefly describe the fix?

SanjaySajuJacob reviewed (approved):
Hey @Uglycrown, the changes look good to me, but would you place take a look at the comments left by @gp201 and merge the latest changes from develop. Thanks!

Uglycrown committed:
Update lesson-information-card-modal.component.spec.ts

Hi @gp201, I combined the two lines you mentioned into a single line, and added a space before every test.

Uglycrown committed:
Merge branch 'develop' into progressbar-read-fix

Uglycrown committed:
Merge branch 'develop' into progressbar-read-fix

Uglycrown commented:
@gp201 PTAL on comments.

Uglycrown committed:
Merge branch 'develop' into progressbar-read-fix

Uglycrown committed:
Update lesson-information-card-modal.component.spec.ts

Uglycrown committed:
Update lesson-information-card-modal.component.spec.ts

Uglycrown reviewed (commented):
@gp201 Removed 75% Test.
Now We have the following tests:
25%
100%
0%
round test

Uglycrown committed:
Merge branch 'develop' into progressbar-read-fix

gp201 commented on core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.spec.ts:
```diff
@@ -436,4 +436,32 @@ describe('Lesson Information card modal component', () => {
+  });
+
+  it('should round down to the nearest whole number', () => {
+    componentInstance.completedCheckpointsCount = 2;
+    componentInstance.checkpointCount = 5;
```
```suggestion
    componentInstance.completedCheckpointsCount = 2;
    componentInstance.checkpointCount = 7;
```

Please update the `progressPercentage` value too.

Uglycrown replied:
done @gp201 PTAL.

gp201 reviewed (changes requested):
1 final nit.

Uglycrown committed:
Merge branch 'develop' into progressbar-read-fix

Uglycrown committed:
Update lesson-information-card-modal.component.spec.ts

Uglycrown reviewed (commented):
@gp201 done the requested changes PTAL.

Uglycrown committed:
Merge branch 'develop' into progressbar-read-fix

Uglycrown committed:
Merge branch 'develop' into progressbar-read-fix

Uglycrown committed:
Merge branch 'develop' into progressbar-read-fix

gp201 reviewed (approved):
LGTM
